pkg/storage/bucket/http/config.go

```diff
@@ -2,20 +2,38 @@ package http
 
 import (
 	"flag"
+	"net/http"
 	"time"
 )
 
+// NOTE some of the fields are hidden in the documentation due to this struct
```

Comment:
nit: remove this and also unhide the settings? these will be added to the docs  by unhiding object_store block with the next minor release which introduces thanos clients